frontend/src/pages/CreateEvent/index.jsx

```diff
@@ -0,0 +1,216 @@
+import React, { useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+
+import { Row, Col, message, DatePicker } from "antd";
+import MainHeader from "../../components/MainHeader";
+import ProfileSider from "../../components/ProfileSider";
+import { Content } from "./style";
+
+import { Select, Form, Input, Divider } from "antd";
+import { FormButton, FormLabel, FormTitle } from "./style";
+
+import { postEvent } from "../../redux/event/api";
+
+import { getTags } from "../../redux/choices/api";
+
+const { Option } = Select;
+
+const CreateEvent = () => {
+
+  // hooks
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const selector = useSelector;
+
+  useEffect(() => {
+    dispatch(getTags());
+    // eslint-disable-next-line
+  },[]);
+
+  // tags that are available
+  const tags = selector((state) => state.choices.tags);
+
+  // initial data of the form
+  const data = {
+    userId: 1,
+    type: "Conference",
+    isPublic: 1,
+    title: "",
+    body: "",
+    link: "",
+    location: "",
+    date: null,
+    other: "",
+    tags: []
+  }
+    
+  // form object
+  const [createEventForm] = Form.useForm();
+
+  // submit action for event creation form
+  const createEventSubmit = function (values) {
+    // get user id
+    var myId = localStorage.getItem("userId")
+
+    // cast date object into string
+    values["date"] = values.date.toString()
+
+    // add user id and visibility
+    values = {
+      ...values,
+      userId: myId,
+      isPublic: 1
+    }
+    dispatch(postEvent(values, history, message));
+  };
+
+  // date validation
+  const validateDate = (rule, value) => {
+    // if null or empty sting there is already another 
+    // error message
+    if(value === "" | value === null){
```

Comment:
interesting choice of `bitwise or` but otherwise functionally unproblematic since used values are booleans